From 23e02ac90627a1ba7a9e099531da762c9712ec51 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 30 Apr 2014 13:27:10 +0800
Subject: [PATCH 62/70] ENGR00311235 Add Ethernet as a valid date connection
 when ota

Enable Ethernet as a valid data connection when ota

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 FSLOta/src/com/fsl/android/ota/OTAServerManager.java | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/FSLOta/src/com/fsl/android/ota/OTAServerManager.java b/FSLOta/src/com/fsl/android/ota/OTAServerManager.java
index 4c0b77f..09dcff3 100644
--- a/FSLOta/src/com/fsl/android/ota/OTAServerManager.java
+++ b/FSLOta/src/com/fsl/android/ota/OTAServerManager.java
@@ -1,5 +1,5 @@
 /*
-/* Copyright 2012-2013 Freescale Semiconductor, Inc.
+/* Copyright 2012-2014 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -89,7 +89,8 @@ public class OTAServerManager  {
 	public boolean checkNetworkOnline() {
 		ConnectivityManager conMgr =  (ConnectivityManager)mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
 		
-		if ((conMgr.getNetworkInfo(ConnectivityManager.TYPE_WIFI)).isConnectedOrConnecting()) {
+		if (conMgr.getNetworkInfo(ConnectivityManager.TYPE_ETHERNET).isConnectedOrConnecting()||
+			conMgr.getNetworkInfo(ConnectivityManager.TYPE_WIFI).isConnectedOrConnecting()) {
 			return true;
 		} else {
 			return false;
@@ -107,7 +108,7 @@ public class OTAServerManager  {
                                 } 
 				else {
 					reportCheckingError(OTAStateChangeListener.ERROR_WIFI_NOT_AVALIBLE);
-                                        Log.v(TAG, "error wifi not avalible");
+                                        Log.v(TAG, "error wifi or ethernet not avalible");
                                 }  
 			}
 			
-- 
1.8.0

